src/BudgetManager/BudgetManager.CQRS/Handlers/WalletHandlers/GetUserWalletTransactionsHandler.cs

```diff
@@ -19,16 +19,16 @@ public GetUserWalletTransactionsHandler(IBaseRepository<User> userRepository, IT
             _transactionRepository = transactionRepository;
         }
 
-        public async Task<WalletTransactionsDTO> Handle(GetUserWalletTransactionsQuery request, CancellationToken cancellationToken)
+        public async Task<WalletTransactionsResponse> Handle(GetUserWalletTransactionsQuery request, CancellationToken cancellationToken)
         {
             User user = await _userRepository.FindByIdAsync(request.userId, cancellationToken);
-            Wallet wallet = user.Wallets.Find(w => w.Id == request.walletId);
+            Wallet? wallet = user?.Wallets?.Find(w => w.Id == request.walletId);
```

Comment:
Why is everything nullable?

Reply:
@Saigunas it's not actually all nullable, I used **?.** - null conditional operator to avoid NullReferenceException rather than using **if** conditions. Because our _user_ could be null if the user with the given id was not found and trying to access Wallets property would result in the exception. Also, the  _Wallets_ property of the user object could also be null, so using _Find_ method with null Wallets would also throw the exception. 
I'm not really sure why I made Wallet object nullable, as reference types could also be null, but doing so eliminated the IDE warning.